### ISSUE TITLE
Metadata: Stackoverflow on unambiguous model with InverseProperty on navigations on both directions

### ISSUE THREAD

**Issue by smitpatel** [aspnet/EntityFramework#7698]
Model:
```C#
//Model1
public class Blog
{
    public int Id { get; set; }
    [InverseProperty(nameof(Post.BlogNav))]
    public List<Post> PostNav { get; set; }
    [InverseProperty(nameof(SpecialPost.BlogInverseNav))]
    public List<SpecialPost> ASpecialPostNav { get; set; }
}

public class Post
{
    public int Id { get; set; }
    [InverseProperty(nameof(Blog.PostNav))]
    public Blog BlogNav { get; set; }
}

public class SpecialPost : Post
{
    [InverseProperty(nameof(Blog.ASpecialPostNav))]
    public Blog BlogInverseNav { get; set; }
}
```
Throws stackoverflow in 1.1.1 packages
```c#
//Model 2
public class Blog
{
    public int Id { get; set; }
    //[InverseProperty(nameof(Post.BlogNav))]
    public List<Post> PostNav { get; set; }
    //[InverseProperty(nameof(SpecialPost.BlogInverseNav))]
    public List<SpecialPost> ASpecialPostNav { get; set; }
}

public class Post
{
    public int Id { get; set; }
    [InverseProperty(nameof(Blog.PostNav))]
    public Blog BlogNav { get; set; }
}

public class SpecialPost : Post
{
    [InverseProperty(nameof(Blog.ASpecialPostNav))]
    public Blog BlogInverseNav { get; set; }
}
```
throws  System.InvalidOperationException: The navigation property 'PostNav' on entity type 'Blog' cannot be associated with foreign key {'BlogId'} because it was created for foreign key {'BlogNavId'}.

**Comment by AndriySvyryd** [comment]
This is similar to https://github.com/aspnet/EntityFramework/issues/7057

**Comment by Eilon** [comment]
This patch fix is *approved*. Please follow the normal code review / pull request process and make sure you make the change in the correct branch.

**Comment by Eilon** [comment]
Preview builds of this patch fix should be available on the following feeds:

* https://dotnet.myget.org/gallery/aspnet-1-0-5-may2017-patch-public (for fixes in the 1.0.x release)
  * Feed URL to add to NuGet.config: https://dotnet.myget.org/F/aspnet-1-0-5-may2017-patch-public/api/v3/index.json
* https://dotnet.myget.org/gallery/aspnet-1-1-2-may2017-patch-public (for fixes in the 1.1.x release)
  * Feed URL to add to NuGet.config: https://dotnet.myget.org/F/aspnet-1-1-2-may2017-patch-public/api/v3/index.json

If you have a chance, please try out these preview builds and let us know if you have any feedback!
